### PR TITLE
This fixes issues 13 and 14

### LINKED ISSUE #13: Improve handling of properties without types
#### Describe the bug
When a property is defined without a type, and it is not an enum, then the template assumes it is a $ref to a different schema and assumes that the title attribute is the class name of the related schema. If there is no $ref then we should raise an exception.

### LINKED ISSUE #14: Extra words show up in the imports section of a schema file.
We saw this:
```
package com.solace.covidtracking.states;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
		has functions.
import java.util.function.Function
```

### PR DESCRIPTION
**Pull request by damaru-inc**
**Description**
This fixes issues 13 and 14, and improves Java formatting of constructors with long arg lists.

It also fixes a bug related to arrays of references to other schemas.

**Related issue(s)**
Fixes #13
Fixes #14

### CODE REVIEW

File: filters/all.js [hunk 1]
```diff
@@ -202,28 +202,47 @@ module.exports = ({ Nunjucks }) => {
...
+
+        if (itemsType === 'object') {
+          isArrayOfObjects = true;
+          itemsType = _.upperFirst(javaName);
+        } else {
+          itemsType = typeMap.get(itemsType);
+        }
       }
       if (!itemsType) {
+<<<<<<< HEAD
```
Comment (mattbishop) [inline filters/all.js:236]: unresolved conflict.

Comment (damaru-inc) [reply filters/all.js]: Oops! Fixed.

File: filters/all.js [hunk 2]
```diff
@@ -202,28 +202,42 @@ module.exports = ({ Nunjucks }) => {
...
     // If a schema has a property that is a ref to another schema,
     // the type is undefined, and the title gives the title of the referenced schema.
     let ret;
     if (type === undefined) {
       if (property._json.enum) {
         ret = _.upperFirst(javaName);
       } else {
-        ret = property.title();
+        // check to see if it's a ref to another schema.
+        ret = property._json['x-parser-schema-id'];
```
Comment (fmvilas) [inline filters/all.js:214]: You can use the public method `ext` or `extension`:

```suggestion
        ret = property.ext('x-parser-schema-id');
```

File: filters/all.js [hunk 3]
```diff
@@ -202,28 +202,42 @@ module.exports = ({ Nunjucks }) => {
...
+          throw new Error("Can't determine the type of property " + name);
+        }
       }
     } else if (type === 'array') {
       if (!property._json.items) {
         throw new Error("Array named " + name + " must have an 'items' property to indicate what type the array elements are.");
       }
-      //console.log('fixtype: ' + JSON.stringify(propery._json.items));
+      //console.log('fixtype: ' + JSON.stringify(property._json.items));
       let itemsType = property._json.items.type;
```
Comment (fmvilas) [inline filters/all.js:225]: I know it's not from this pull request but try to never access `_json` directly. Instead, use the public methods:

```suggestion
      let itemsType = property.items().type();
```

File: filters/all.js [hunk 4]
```diff
@@ -202,28 +202,42 @@ module.exports = ({ Nunjucks }) => {
...
+          isArrayOfObjects = true;
+          itemsType = _.upperFirst(javaName);
+        } else {
+          itemsType = typeMap.get(itemsType);
+        }
       }
       if (!itemsType) {
-        itemsType = _.upperFirst(javaName);
-        isArrayOfObjects = true;
+        itemsType = property._json.items['x-parser-schema-id'];
```
Comment (fmvilas) [inline filters/all.js:236]: Same here.

Comment (damaru-inc) [reply filters/all.js]: Okay, I think I've fixed them all.

**Pull request merged**